lgc/patch/LowerGpuRt.cpp

```diff
@@ -117,26 +118,48 @@ unsigned LowerGpuRt::getWorkgroupSize() const {
 // Get flat thread id in work group/wave
 Value *LowerGpuRt::getThreadIdInGroup() const {
   // Todo: for graphics shader, subgroupId * waveSize + subgroupLocalInvocationId()
-  unsigned builtIn = m_context->getPipelineType() == PipelineType::Graphics ? lgc::BuiltInSubgroupLocalInvocationId
-                                                                            : lgc::BuiltInLocalInvocationIndex;
-  lgc::InOutInfo inputInfo = {};
-  return m_builder->CreateReadBuiltInInput(static_cast<lgc::BuiltInKind>(builtIn), inputInfo, nullptr, nullptr, "");
+  unsigned builtIn = m_pipelineState->isGraphics() ? BuiltInSubgroupLocalInvocationId : BuiltInLocalInvocationIndex;
+  InOutInfo inputInfo = {};
+  return m_builder->CreateReadBuiltInInput(static_cast<BuiltInKind>(builtIn), inputInfo, nullptr, nullptr, "");
 }
 
 // =====================================================================================================================
 // Create global variable for the stack
-void LowerGpuRt::createGlobalStack() {
-  auto ldsStackSize = getWorkgroupSize() * MaxLdsStackEntries;
-  // Double anyhit and intersection shader lds size, these shader use lower part of stack to read/write value
-  if (m_lowerStack)
-    ldsStackSize = ldsStackSize << 1;
-
-  m_stackTy = ArrayType::get(m_builder->getInt32Ty(), ldsStackSize);
-  auto ldsStack = new GlobalVariable(*m_module, m_stackTy, false, GlobalValue::ExternalLinkage, nullptr,
-                                     RtName::LdsStack, nullptr, GlobalValue::NotThreadLocal, 3);
-
-  ldsStack->setAlignment(MaybeAlign(4));
-  m_stack = ldsStack;
+// @param [in/out] module : LLVM module to be run on
+void LowerGpuRt::createGlobalStack(Module &module) {
+
+  struct Payload {
+    bool needGlobalStack;
```

Comment:
If it is not against common practice, I'd prefer to initialize the members inside the struct

Reply:
I do see some other places use this form, e.g.
https://github.com/GPUOpen-Drivers/llpc/blob/8e2673136e10557738b047dd1b8584dde1ba9590/lgc/patch/PatchEntryPointMutate.cpp#L271